functions/lib/github.js

```diff
@@ -1,167 +1,202 @@
-      // the identities associated with all commits of this PR.
-      const identities = getPrIdentities(pr, ghClient)
+    // If any error occurred, improve user experience by posting a comment.
+    if (status.state === 'error' && !status.comment) {
+      status.comment =
+        `Unable to verify CLA: ${status.description}. ` +
+        'If the problem persists, please contact support@opennetworking.org ' +
+        ` (\`support-id: ${eventSnapshot.id}\`)`
+    }
 
-      return clalib.checkIdentities(identities).then(result => {
-        if (result.allWhitelisted) {
-          console.log('cla is signed for all commits')
-          status.state = 'success'
-          status.description = 'All good! We have a CLA in file for all contributors in this PR.'
+    // Post status to Github and update event in the DB
+    const statusPromise = octokit.repos
+      .createStatus({
+        owner: pr.base.repo.owner.login,
+        repo: pr.base.repo.name,
+        context: 'onf/cla',
+        sha: pr.head.sha,
+        target_url: 'https://cla.opennetworking.org',
+        description: status.description,
+        state: status.state
+      })
+      .then(() => {
+        status.githubAck = true
+      })
+      .catch(error => {
+        status.githubAck = false
+        status.githubError = JSON.parse(JSON.stringify(error))
+      })
+      .then(() => {
+        event.status = status
+        return eventSnapshot.ref.update(event)
+      })
+      .catch(console.error)
+
+    const contribRef = db.collection('contributions')
+      .doc(event.contributionId)
+    const commentPromise = contribRef.get()
+      .then(snapshot => {
+        // Retrieve existing comment_id (if any)
+        if (!snapshot.exists) {
+          console.error(`Missing contribution ${event.contributionId} in DB`)
+          return null
         } else {
-          let msg
-          if (result.missingIdentities.length) {
-            msg = 'We could not find a CLA for the following identities: ' +
-              result.missingIdentities +
-              '. You will need to sign one before we can merge your PR.'
+          return snapshot.data().githubCommentId
+        }
+      })
+      .then(commentId => {
+        const commentData = {
+          owner: pr.base.repo.owner.login,
+          repo: pr.base.repo.name,
+          issue_number: pr.number
+        }
+        if (status.comment) {
+          // Create or update existing comment.
+          commentData.body = status.comment
+          if (commentId) {
+            commentData.comment_id = commentId
+            return octokit.issues.updateComment(commentData)
           } else {
-            msg = 'We were not able to verify the CLA for this PR. ' +
-              'If the problem persists please contact support@opennetworking.org'
+            return octokit.issues.createComment(commentData)
+              .then(response => contribRef.update({
```

Comment:
do not nest Promises, you can  `return octokit.issues.createComment(commentData)` and add a `.then`

Reply:
Done